### ISSUE TITLE
One channel somehow disappeared

### ISSUE THREAD

**Issue by ssergiienko** [raelgc/scudcloud#454]
Maybe some slack changes breaks something. In native slack linux client all works fine.
New messages that appears in this channel successfully appears in system notifications, but I can't find this channel on left panel

P.S. Archlinux:
local/scudcloud 1.24-1 (messaging)
local/qt4 4.8.7-9
local/qt5-base 5.6.0-7 (qt qt5)

**Comment by raelgc** [comment]
Hi @ssergiienko, thanks for raising this issue.

Can you, please, include a screenshot?

**Comment by ssergiienko** [comment]
Hi @raelgc, is this what you need? Feel free to ask more info, if it helps

![screenshot_20160611_093043](https://cloud.githubusercontent.com/assets/9850672/15983640/513bd84e-2fb7-11e6-8cc8-984627253680.png)

Native slack client:
![screenshot_20160611_093126](https://cloud.githubusercontent.com/assets/9850672/15983641/513f7d14-2fb7-11e6-8720-12a39dde591d.png)

I've tried various "Channel List" preperences but it didn't help

As you may noticed, there is different number of channels on this screenshots, I've tried to find "#general" with filter in "Browse channels" and Scudcloud didn't see it, but Slack native does and shows that I'm joined, so this is probably not a displaying issue

**Comment by cjunge-work** [comment]
I also have this issue. I have 2 accounts, 1 account shows #general, the other does not. It's not even available in "browse all channels".

**Comment by ssergiienko** [comment]
Something strange happens: after someone add new user to "#general" channel it appears in Scudcloud, but after restart - channel disappeared again.

**Comment by neokore** [comment]
Same here. Using Manjaro with KDE.

**Comment by pdecat** [comment]
Same here, Ubuntu 16.04:

```
# apt-cache policy scudcloud 
scudcloud:
  Installed: 1.24-2
  Candidate: 1.24-2
  Version table:
 *** 1.24-2 500
        500 http://ppa.launchpad.net/rael-gc/scudcloud/ubuntu xenial/main amd64 Packages
        500 http://ppa.launchpad.net/rael-gc/scudcloud/ubuntu xenial/main i386 Packages
        100 /var/lib/dpkg/status
```

**Comment by raelgc** [comment]
This is really weird. But I suspect it's related to a CSS fix we tried some weeks ago.

Can someone with this issue, please, edit file `scudcloud/resources/resources.css`, remove the last line, save it and restart ScudCloud?

ScudCloud folder is usually located at `/usr/lib/python3/dist-packages/` or `/usr/lib/python3-4/dist-packages/`

**Comment by pdecat** [comment]
Hi @raelgc, I commented the last line and restarted scudcloud but this did not help.

Please also note that the channels count is also wrong: in my case 10 with scudcloud instead of 11 with the web version.

**Comment by neokore** [comment]
Yes @raelgc , deleting the '.overflow_ellipsis' rule makes the 'general' channel appears again.
Also, my channel count is now right (22 on both).

**Comment by ssergiienko** [comment]
@raelgc Tied too, no success

```
/* Force monospaced fonts in <pre> (Fixes #428) */
pre, pre > span { font-family: Monaco,Menlo,Consolas,"Courier New",monospace; }
/* Assign .overflow_ellipsis width (Fixes #437) */
/* .overflow_ellipsis { width: 148px; } */      # I've commented this one
```

Didn't helps.

+1 About wrong channels count, so this is not a display issue

**Comment by raelgc** [comment]
@pdecat and @ssergiienko  Read more about channel limit in #376. Unfortunately we have no control over this.

**Comment by ssergiienko** [comment]
@raelgc Thanks for help!
I've tried to resize window twice as larger then my screen and this changes nothing. Also, it's strange why all worked perfectly with ~35-38 channels for a long time and now limits becomes a problem(

**Comment by pdecat** [comment]
This is strange. Whatever the resolution, even full screen in 4k, I still see a 10 channels count. Also, searching channels is also affected, I cannot find the #general channel in scudcloud for the affected team.

And I've got another team with 71 channels correctly displaying in scudcloud.

**Comment by pdecat** [comment]
Here are screenshots to better explain the issue.

Scudcloud:
![Scudcloud](https://cloud.githubusercontent.com/assets/318490/16041690/2815cfb0-3237-11e6-97e4-b18b9c3fecfe.png)

Same team in web browser:
![Web](https://cloud.githubusercontent.com/assets/318490/16041694/2b3d7ac6-3237-11e6-93ac-1caa19907260.png)

**Comment by pdecat** [comment]
I created an additional _test1_ channel in the affected team and counts are now 11 and 12 respectively in the scudcloud app and the web browser.

**Comment by pdecat** [comment]
Launching `scudcloud --debug True`, I can verify in the webkit console that the _general_ channel is there but somehow not bound or recognized as such:

```
> TS.channels.getChannelByName("general")
Object {id: "C0K9HRAP7", name: "general", is_channel: true, created: 1453709306, creator: "U0K9HRALV"…}

> TS.channels.getGeneralChannel()
undefined
```

In Chrome, both methods return the same channel object:

```
> TS.channels.getChannelByName("general")
Object {id: "C0K9HRAP7", name: "general", is_channel: true, created: 1453709306, creator: "U0K9HRALV"…}

> TS.channels.getGeneralChannel()
Object {id: "C0K9HRAP7", name: "general", is_channel: true, created: 1453709306, creator: "U0K9HRALV"…}
```

**Comment by raelgc** [comment]
If we got able to reproduce this consistently, we can fix it.

Just to add to the discussion: in the latest release I just changed the CSS to change the team pane width, because long team names were breaking the layout. No changes to the internals.

Scudcloud itself is a simple wrapper for the web version, like the official app. How both app works: Slack will read some JS files that we provide, and then will behave by itself.

The main differences: we use a qtwebkit engine only, while the official app is using elektron (which is Chromium), and, of course, they know all the Slack internals and settings.

Or JS is at https://github.com/raelgc/scudcloud/blob/master/sources/scudcloud-src.js

I'm thinking if we can change any preferences to make all teams visible..

**Comment by pdecat** [comment]
@raelgc I'm a newcomer to scudcloud, only been using for two days. Great work by the way!

So I never experienced any previous version and cannot tell if it is a regression or not.

**Comment by raelgc** [comment]
@pdecat This issue is new. And thank you for all the feedback provided!

**Comment by cwoolner** [comment]
Having the same issue for all of my slack groups.  Started happening (or at least noticed) on June 12th.  Commenting-out the CSS didn't change anything.

Only thing to add to what has already been said is that I get thrown into #general when I initially log in and I see all my messages as expected, but it's not in my channel list.  When I switch to another channel, I then can't get back to #general.  As others have said, it's not in my Ctrl+k list either.

Running Ubuntu 16.04.

Thanks!

**Comment by fabiofranco85** [comment]
Hi all, just to say that the same problem is happening with me too... running Debian 8.5 + KDE 4.14.2 + ScudCloud 1.24
Thanks!

**Comment by stevenjkern** [comment]
I haven't had a chance to set up a working dev environment to look into it myself yet, but my guess would be that Slack has stopped providing the general channel when Scudcloud calls to get the list of channels [here](https://github.com/raelgc/scudcloud/blob/master/scudcloud/wrapper.py#L178).

**Comment by raelgc** [comment]
@stevenjkern This method is only used to display a channel list in Unity quick list (when you right click an app icon in Unity launcher).

ScudCloud do not render directly any Slack UI other then the left (teams) pane.

**Comment by stevenjkern** [comment]
@raelgc, ah I see. Thanks for the clarification.

**Comment by stevenjkern** [comment]
Just logged in, and general is back for me. :tada:

**Comment by cmuell89** [comment]
I have this issue as well. Definitely hoping it gets fixed as slack's "beta" app for linux shits the bed constantly. Scudcloud's stability is great but this is a very irksome bug as it forces me to use the slack webapp. Commenting out the last line in resources.css did not work for me.

**Comment by jasonkuehl** [comment]
So I too have this issue. But I kind of like General missing...

But I can confirm I do as well have this issue.
scudcloud:
  Installed: 1.24-2
  Candidate: 1.24-2
  Version table:
 **\* 1.24-2 500
        500 http://ppa.launchpad.net/rael-gc/scudcloud/ubuntu xenial/main amd64 Packages
        500 http://ppa.launchpad.net/rael-gc/scudcloud/ubuntu xenial/main i386 Packages
        100 /var/lib/dpkg/status

1:630)$ uname -a
Linux StarKillerBase 4.4.0-24-generic #43-Ubuntu SMP Wed Jun 8 19:27:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

Theme is cinnamon 

![selection_018](https://cloud.githubusercontent.com/assets/9578513/16089603/ed1af610-32f9-11e6-996b-9541390b3540.png)

**Comment by jasonkuehl** [comment]
Im seeing this with other channel names as well, not sure General. I can join the channel but as soon as I leave the channel its goes away.

**Comment by pdecat** [comment]
The issue is gone for me, everything is back to normal in all my slack teams.

Must have been an issue on the slack side...

Edit: spoke too fast, the issue is back on one team after a restart :(

**Comment by cjunge-work** [comment]
I just restarted Scudcloud and my issue is fixed.

**Comment by cwoolner** [comment]
Restarting didn't seem to work for me earlier in the day, but a quick reinstall did the trick.  Things seem to be back to normal at the moment.

**Comment by raelgc** [comment]
I'm facing this issue too in a team with more than 10 channels (21 channels). I can see `#general` available on start, but after some activity, it goes away.

**Comment by raelgc** [comment]
Guys, can you please, try this?

Edit `/usr/lib/python3/dist-packages/scudcloud/__main__.py` and change this line (usually line `42`):

``` python
    app.setApplicationName(Resources.APP_NAME)
```

by this?

``` python
    app.setApplicationName('ScudCloud Slack_SSB')
```

And, of course, restart ScudCloud.

**Comment by jayme-github** [comment]
Changing `ApplicationName` brought `#general` back again but that could also be related to the restart of cause. Will report back in the next days.

**Comment by qristin** [comment]
I had the same problem and this helped indeed.

**Comment by ssergiienko** [comment]
For me, bug disappeared without any modifications) No fix was needed

**Comment by raelgc** [comment]
Indeed, like we said before, this is an internal Slack behavior: it started without code changes from our side, probably can stop without code changes too.

To explain the name change attempt: the internal Slack JS engine uses different methods to guess if it's running in a browser or desktop app. In few places (I know this is a bit weird) it's the name.

Without change app name, after restart, I can see `#general`, but it disappears after few minutes.

Changing the name put `#general` back and displayed a 15 teams in the teams panel instead of only 10.

**Comment by raelgc** [comment]
Ok, I'm publishing new versions in the PPA with this fix. In fact, we already used this internal name in the past.

Let me re-open the issue (closed automatically due the merge) in case more users arrive seeking for a fix.

**Comment by pdecat** [comment]
Just upgraded, everything's ok.
Thanks!

**Comment by jasonkuehl** [comment]
Confirmed working.

Thanks again @raelgc

**Comment by raelgc** [comment]
You're welcome. Thank you guys, for all feedback and patience and sorry for the inconvenience.

**Comment by raelgc** [comment]
No more reports of this issue.

**Comment by jonnybazookatone** [comment]
I believe I have the latest scudcloud, v1.26, running on Ubuntu 14.04. I still have intermittent behaviour for the general channel. Today it's not there, but yesterday is was, much like other users above. Any suggestion welcome, I might be doing something stupid on my side.

**Comment by raelgc** [comment]
@jonnybazookatone I faced something similar today again: one specific channel that I used more than `general` just disappeared (and my count of channels got decrease by one).

The workaround was right click and Reload the page to get the channel back.

**Comment by cwoolner** [comment]
Yes same thing started happening again, but only for one of my Slack groups so far.

Right click -> Reload workaround worked.  Thanks for the tip!

**Comment by shevek** [comment]
Reload workaround NOT working for me.

**Comment by shevek** [comment]
If I "go into" a different channel, THEN reload, then I see #general, but NOT the channel I'm now "in". So clearly I'm allowed to see all channels except my "current" one.

**Comment by dekhtyarev** [comment]
I have is observed the same behavior as that of @shevek

```
$ cat /etc/issue
Ubuntu 16.04 LTS
$ dpkg -l | grep scudcloud
ii  scudcloud                                       1.26-2                                     all          ScudCloud is a non official desktop client for Slack
```

**Comment by fschrofner** [comment]
Also experiencing the same behaviour as @shevek 
I still have the 1.25 release, though

```
flosch at flosch-elitebook in ~
❯ uname -r                                                                                                                                                                               13:38
4.6.2-1-ARCH

flosch at flosch-elitebook in ~
❯ ls -la /var/lib/pacman/local | grep scudcloud                                                                                                                                          13:38
drwxr-xr-x    2 root root  4096 Jun 27 13:38 scudcloud-1.25-1

```

It works in [Franz](http://meetfranz.com/) (an Electron based client) and the webclient, however.

**Comment by Na3blis** [comment]
Still experiencing the issue on 1.26-2 unfortunately

**Comment by DexterTheDragon** [comment]
I'm having a similar thing to @shevek. When I right click -> Reload, whatever channel I'm currently in disappears from the list. However, if I select a Direct Messages room and do right click -> Reload then all channels appear and so far have stayed.

Also, when I go to File -> Preferences clicking on Advanced Options doesn't work. Not sure if that's related or a different problem.

**Comment by skelband** [comment]
I also experience this issue with 1.26-2. #general is not listed in the channel list.
I have 4 other channels in the list, but #general is not there. 
Restarting does not help.
Using official slack client shows the channel as normal.
Can I try anything to diagnose this?

**Comment by skelband** [comment]
I also find that Advanced Options is not responsive.

**Comment by skelband** [comment]
Ah yes. As for @DexterTheDragon above, the reload did the trick. Will monitor the situation.

**Comment by skelband** [comment]
OK more information. If I click on "Reload" for CHANNELS it #general appears. If I click on "Reload" for DIRECT MESSAGES, it disappears from the CHANNELS list. Weird.

**Comment by bstein-clever** [comment]
I also experienced this last week. Worse, I was unable to even get to the channel at all, as it did not appear in the "jump to a conversation" dialog window (Ctrl-K).

Restarting Scudcloud did not help, but I tried leaving a few different channels, and restarting did bring it back after that. I'm not sure if it was some sort of weird limit with number of open channels - it was probably one of the first channels created by our team, so likely has a low Channel ID...

**Comment by nethershaw** [comment]
I had been running Scudcloud 1.16 for months without any issues, and this problem seemed to spontaneously appear a few weeks ago. It's still happening on 1.25.

I'm signed in to three teams, and I am in fewer than six channels on each team. The channel that goes 'missing' always seems to be whichever one gets displayed first or by default (typically #general, but not always) when I sign in. All teams are almost always missing exactly one channel in which I am present, and reloading (Ctrl+R) makes the channel I am currently viewing disappear from the list while restoring any other channels that might be 'missing.'

If it's relevant, I'm on Gentoo, using Python 3.4, and Gnome 3. I cobbled together my own ebuild for the app so it's properly installed, provided here in case it is helpful to anyone: https://github.com/nethershaw/aptenodytes/blob/master/net-im/scudcloud/

**Comment by raelgc** [comment]
I'm really sad this is happening. And, this issue appears to be affecting all of us using ScudCloud.

So, please, post new comments only if find new info about it. Avoid "me too".

And, just to clarify again: ScudCloud is a simple wrapper around the webversion (like all Slack clients). There is no messaging API, and we're not touching channel list or any other internals.

My guess is that Slack improved their internal JS, and due the old webkit (the webengine ScudCloud uses) version used in our distros, we're facing this kind of issues.

So, still using python we have 2 solutions: test with a newer version of webkit, or try use the webviewengine - a more recent engine based on Chromium, like everyone is doing today.

Two downsides here: more RAM usage and it'll require some Qt package installation, which used to be a kind of big task to solve.

**Comment by SevereOverfl0w** [comment]
http://webkitgtk.org/ I'm not sure if this is worth considering, there are python bindings https://pypi.python.org/pypi/pywebkitgtk/1.8 and it appears to have been upated lately.

It may not be any more up to date than qtwebkit, but something else to consider depending on the community goals.

My personal goals align with having a lightweight way to run slack.

**Comment by Josaphat** [comment]
Can confirm that reloading does "work around" the problem in a way, but it makes whatever channel I was on when reloading disappear.  My workaround has been to create a dummy "#workaround" channel, select that, and then reload.

**Comment by raelgc** [comment]
Current workaround: click to reload inside any direct message conversation.

**Comment by raelgc** [comment]
The solution, I think, it will rely in some JS configuration. We can try start ScudCloud with `scudcloud --debug=True`, right click, Inspect, maximize inspect window and select Console tab.

In there, we have the `TS` object, which is the Slack core. 

Now I'm trying to compare the same channel in Slack in the browser with the same channel running in ScudCloud. You can get the channel object using:

``` javascript
TS.channels.getChannelByName("<your-channel-name>")
```

Of course, replacing `<your-channel-name>` by the name of the channel.

**Comment by bstein-clever** [comment]
Except ... I don't think it's just that the channel isn't getting displayed.
When it's missing, I can't even select it from the "Jump to a conversation" window when I type out the channel name explicitly.

**Comment by davidwindell** [comment]
Also, I can't search for the channel in the available channels list!?

Could it be to do with reserved names or something? #team and #general?

**Comment by neokore** [comment]
This may be an error when getting current channel: the behaviour I found is that when I reload Scudcloud with a channel (or a multiuser direct messaging chat) selected, this channel will disappear; but, if I do it with a (single) user direct messaging chat selected, all channels will appear. That's why this bug appears randomly.

I didn't take a look at the code yet, but maybe a possible workaround is to load the channel list before getting current channel.

Hope it helps!

**Comment by raelgc** [comment]
@neokore Sounds like the current channel should be added to a list of active channels and it's not.

**Comment by keevitaja** [comment]
Why was this closed? #466 is closed as well!

**Comment by SevereOverfl0w** [comment]
Just ran the latest commit https://github.com/raelgc/scudcloud/commit/6d69f777ff95c85017f57eb4c95003e0d77dd039 works brilliantly. Thanks @raelgc!

**Comment by raelgc** [comment]
Reopening (it was closed automatically by commit) to keep people aware of this.

Ubuntu/Debian versions are published. In the next hours probably we'll see for other distros too.

**Comment by ssergiienko** [comment]
Archlinux: works fine (v1.28)
Thanks!

**Comment by raelgc** [comment]
Thank you guys, for the patience and all feedback provided!

**Comment by KlipperKyle** [comment]
I can confirm that the #general channel shows up in v1.28.  No reloading or other ritual dances are needed.

Platform: Arch Linux x86_64, scudcloud AUR package built in a clean chroot using extra-x86_64-build.

**Comment by nethershaw** [comment]
Thank you, @raelgc. I've updated the ebuild in my Gentoo overlay.

**Comment by skelband** [comment]
I also confirm fix in 1.29. Many thanks!

**Comment by shevek** [comment]
Fixed for me in 1.29-2.

**Comment by mattdowle** [comment]
Also had this problem.  Found this issue, upgraded to your PPA and it's fixed.
Many many thanks @raelgc !
